src/iov.c

```diff
@@ -79,8 +82,11 @@ void ofi_consume_iov_desc(struct iovec *iov, void **desc,
 			sizeof(*desc) * (*iov_count - i));
 	*iov_count -= i;
 out:
-	iov[0].iov_base = (uint8_t *)iov[0].iov_base + to_consume;
-	iov[0].iov_len -= to_consume;
+	if (to_consume < iov[0].iov_len) {
```

Comment:
nit: if the if statement needs braces, then add them to the else statement.

Btw, the code is optimized for the case ofi_total_iov_len() > to_consume.  It works if it's equal, but ends up calling memmove(..., 0), iov_base + 0, and iov_len - 0.

At least with the tcp provider, this function is only called if iov_len > to_consume.  Otherwise, it's skipped.  I'm not sure about EFA or other use cases, such that we may want to add an optimization for it (at the expense of adding another branch to the current optimized case.

Reply:
Addressed the nit and added protection for memmove.